besu/src/main/java/org/hyperledger/besu/cli/util/CommandLineUtils.java

```diff
@@ -51,20 +51,59 @@ public static void checkOptionDependencies(
       final boolean isMainOptionCondition,
       final List<String> dependentOptionsNames) {
     if (isMainOptionCondition) {
-      final String affectedOptions =
-          commandLine.getCommandSpec().options().stream()
-              .filter(
-                  option ->
-                      Arrays.stream(option.names()).anyMatch(dependentOptionsNames::contains)
-                          && !option.stringValues().isEmpty())
-              .map(option -> option.names()[0])
-              .collect(
-                  Collectors.collectingAndThen(
-                      Collectors.toList(), StringUtils.joiningWithLastDelimiter(", ", " and ")));
+      final String affectedOptions = getAffectedOptions(commandLine, dependentOptionsNames);
 
       if (!affectedOptions.isEmpty()) {
         logger.warn(DEPENDENCY_WARNING_MSG, affectedOptions, mainOptionName);
       }
     }
   }
+
+  /**
+   * Check if options are passed that require an option to be true to have any effect and log a
+   * warning with the list of affected options. Multiple main options may be passed to check
+   * dependencies against.
+   *
+   * <p>Note that in future version of PicoCLI some options dependency mechanism may be implemented
+   * that could replace this. See https://github.com/remkop/picocli/issues/295
```

Comment:
is anything happening with 295?

Reply:
It's been implemented, but it'd require rework and I don't think it addresses this issue:
https://picocli.info/#_argument_groups